src/mesido/workflows/emerge.py

```diff
@@ -0,0 +1,246 @@
+        ----------
+        source : string of the source name that is going to be minimized
+        """
+        # self.target_max = 0
+        # self.function_range = (-1.0e9, 1.0e9)
+        # self.function_nominal = 1.0e7
+
+        self.asset_name = asset_name
+
+    def function(
+        self, optimization_problem: CollocatedIntegratedOptimizationProblem, ensemble_member: int
+    ) -> ca.MX:
+        """
+        This function returns the state variable to be minimized.
+
+        Parameters
+        ----------
+        optimization_problem : The optimization class containing the variables'.
+        ensemble_member : the ensemble member.
+
+        Returns
+        -------
+        The negative hydrogen production state of the optimization problem.
+        """
+        # TODO: not yet scaled
+        return -optimization_problem.extra_variable(f"{self.asset_name}__revenue", ensemble_member)
+
+
+class MinCost(Goal):
+
+    priority = 1
+
+    order = 1
+
+    def __init__(self, asset_name: str):
+        # self.target_max = 0.0
+        # self.function_range = (0.0, 1.0e9)
+        # self.function_nominal = 1.0e7
+
+        self.asset_name = asset_name
+
+    def function(
+        self, optimization_problem: CollocatedIntegratedOptimizationProblem, ensemble_member: int
+    ) -> ca.MX:
+
+        return optimization_problem.extra_variable(
+            f"{self.asset_name}__fixed_operational_cost", ensemble_member
+        ) + optimization_problem.extra_variable(
+            f"{self.asset_name}__variable_operational_cost", ensemble_member
+        )
+
+
+class EmergeWorkFlow(
+    ESDLAdditionalVarsMixin,
+    TechnoEconomicMixin,
+    LinearizedOrderGoalProgrammingMixin,
+    SinglePassGoalProgrammingMixin,
+    ESDLMixin,
+    CollocatedIntegratedOptimizationProblem,
+):
+    """
+    This problem class is for the absolute heat tests. Meaning that this problem class
+    is applied to an esdl where there is no dedicated supply or return line. For this test case
+    we just match heating demand (_GoalsAndOptions) and minimize the energy production to have a
+    representative result.
+    """
+
+    # TODO: this workflow is still work in progress and this part of code still needs to be
+    #  finalised
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.gas_network_settings["head_loss_option"] = HeadLossOption.NO_HEADLOSS
+
+    def goals(self):
+
+        goals = super().goals().copy()
+
+        for asset_name in [
+            *self.energy_system_components.get("electricity_demand", []),
+            *self.energy_system_components.get("gas_demand", []),
+        ]:
+            goals.append(MaxRevenue(asset_name))
+            goals.append(MinCost(asset_name))
+
+        for asset_name in [
+            *self.energy_system_components.get("electricity_source", []),
+            *self.energy_system_components.get("gas_tank_storage", []),
+            # TODO: battery
```

Comment:
add the battery (electricity_storage) it is already in the constraints

Reply:
done